### DISCUSSION TITLE
Option to takeover existing stream instead of blocking

### DISCUSSION THREAD

**Discussion by michellabbe** [LizardByte/community#437] [Sunshine]
Sunshine rocks as a nVidia Gamestream replacement.

When stream is already open and trying to connect from another computer, nVidia's behavior is to takeover the existing session from the new computer and disconnect the old client.

Sunshine's behavior is to display an error message and force us to manually stop existing stream, which is fine for most people who always stream from the same device and/or don't want their active stream disturbed while playing.

However, it can quickly become an annoyance to have to walk the other room or use VNC to stop stream when your kids move often between multiple client devices (e.g. nVidia Shield/Chromecast on big TV without keyboard/mouse, computer next to dad's, or main pc in kids room) and/or forgot again to stop stream before quitting Moonlight.

Adding an option to change Sunshine's behavior would be an easy way to get rid of this annoyance in those situations.

**Comment by ReenigneArcher** [comment]
I think if you increase "channels" by 1 it will solve your problem.

**Comment by michellabbe** [comment]
Thanks for your reply.

I saw that suggestion in the error message window, but my understanding is that will produce a second simultaneous stream and make cpu work harder, which is not desired.

I will test it though, if it's like 0.5% harder it's no big deal.

**Reply by ReenigneArcher** [reply]
When you are not actively streaming, the stream is paused and "should" not consume much extra CPU at all, if any. But you need an additional channel to allow a second device to connect... You'll still have only one active stream so I don't believe your CPU resources will be consumed unless you actually connect a second simultaneous stream.
